assets/js/editor/editor.js

```diff
@@ -614,19 +621,87 @@ class Communicator {
     };
     const response = await fetch(SOLUTIONS_EDITOR_URL, requestConfig);
     if (response.status !== 200) {
-      alert(getString(msgs.error_saving_files));
+      showAlert(getString(msgs.error_saving_files));
       return;
     }
     const data = await response.json();
     window.location.assign(data.success ? SOLUTIONS_LIST_URL : SOLUTIONS_EDITOR_URL);
   }
+
+  async checkSyntax() {
+    if (!SYNTAX_CHECK_URL) return;
+    const payload = {
+      files: fileBuilder.files.map((file) => {
+        return { name: file.fileName, contents: file.fileContent };
+      }),
+    };
+    const requestConfig = {
+      method: "POST",
+      headers: {
+        "X-CSRFToken": CSRF_TOKEN,
```

Comment:
CSRF token is not needed for POSTs to the syntax check service, because the real one is external to the Django app. (The mock endpoint simulates this, too).